src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,95 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private static final double GROW_FACTOR = 1.5;
+    private T[] elementData = (T[]) new Object[INITIAL_CAPACITY];
+    private int size;
+
     @Override
     public void add(T value) {
-
+        ensureArrayCapacity();
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        ensureArrayCapacity();
+        checkIndex(index, size);
```

Comment:
I'd swap this lines in order not to grow array if we're going to abort operation due to index
```suggestion
        checkIndex(index, size);
        ensureArrayCapacity();
```